src/store/modules/collection.ts

```diff
@@ -17,7 +19,20 @@ class Collection extends VuexModule {
   public page = 0;
 
   get getPosts() {
-    return this.items?.filter($ => $.type in PostTypes);
+    if (this.items) {
+      const postTypeItems = this.items.filter($ => $.type in PostTypes);
+
+      const activityItems = this.items
+        .filter($ => !($.type in PostTypes))
+        .filter(
+          ($: Activity) =>
+            !!$.object &&
+            ActivityObjectHelper.hasProperty($.object, "type") &&
+            ($.object as ActivityObject).type in PostTypes
+        )
```

Comment:
this looks somehow like duplicate code for me (line 26 vs line 31). What's the idea here?